pkg/util/template/resource.go

```diff
@@ -30,11 +31,27 @@ type HTML struct {
 	IsFindAllowedInFs FindResourcesFsFilter
 }
 
+type CustomizableHTML struct {
```

Comment:
I took a closer to the code, I realize that we have one `HTML` descriptor for each template. I was mistaken that one HTML `descriptor` manages all HTML templates. Sorry for confusing you.

here is the picture in my mind.

```golang
// HTML defines a HTML template that CANNOT be customized by the app.
type HTML {
	// Name is the name of template
	Name string
	// ComponentDependencies is the HTML component templates this template depends on.
	ComponentDependencies []*HTML
        // IsFindAllowedInFs is removed. The existing behavior of ExcludeAppFs should be embodied into this descriptor. Please also remove FindResourcesFsFilter
}

// MessageHTML represents a message in HTML. It can be customized by the app.
type MessageHTML struct {
	// Name is the name of template
	Name string
        // Omit ComponentDependencies. We do not need it in this descriptor
}
```

- MatchResources is the same as before. You can just duplicate it for HTML and MessageHTML.
- FindResources of HTML is ExcludeAppFs. Find FindResources of MessageHTML is AnyFs.
- ViewResources is  the same as before. You can just duplicate it for HTML and MessageHTML.
- UpdateResource of HTML always results in an error. UpdateResource of MessageHTML is almost the same as before, with an addition of respecting the new feature config.
- Repeating the same thing for PlainText and MessagePlainText.

Reply:
updated, plz check